### PR TITLE
Properly handle Dutch year numbers

### PR DESCRIPTION
**Pull request by mvz**
Changes the handling of, e.g., 1984, to use nineteenhundred-style instead of one thousand nine hundred.

This solution is slightly hacky at the moment. If this logic makes sense for other languages (English and German, perhaps?) I can create some helper methods.

### CODE REVIEW

File: lib/numbers_and_words/strategies/figures_converter/languages/nl.rb [hunk 1]
```diff
@@ -24,6 +24,19 @@ def simple_number_to_words
...
+            if figures.number_in_capacity(1) == 1
+              if figures.hundreds
+                teen_hundreds = @figures[2, 2].to_figures
+                [([@translations.teens(teen_hundreds), translate(:hundreds, 1)] + simple_number_to_words).join]
+              else
+                simple_number_to_words + [translate(:mega, 1)]
+              end
+            else
+              super
+            end
```
Comment (jlduran) [inline lib/numbers_and_words/strategies/figures_converter/languages/nl.rb:38]: This is just a personal preference, how about?:

```ruby
return super unless figures.number_in_capacity(1) == 1

if figures.hundreds
  teen_hundreds = @figures[2, 2].to_figures
  [([@translations.teens(teen_hundreds), translate(:hundreds, 1)] +
    simple_number_to_words).join]
else
  simple_number_to_words + [translate(:mega, 1)]
end
```

Comment (mvz) [reply lib/numbers_and_words/strategies/figures_converter/languages/nl.rb]: Yes, that's better. Updated.

**Comment by mvz** [comment]
The teenhundreds style is generally used with years, not so much with other numbers. Still to do:
- [ ] Use this also for later years, like 2183
- [ ] Think of a better option name since this one is bad for 2183

**Comment by jlduran** [comment]
> Maybe `:years` or `:year_numbers` could be a good option name?

Yes, if I understand what this commit is about. Based on (US) English:

- 1,984 should be spelled out by default: _one thousand nine hundred eighty-four_.
- 1984 plus an option (could be `:years`) could be spelled out: _nineteen eighty-four_. I believe this option may have its uses in some sort of text-to-speech application, as not many manuals of style recommend it.
- 1,984 plus an option (could be `:keep_spelling_hundreds_instead_of_thousands` :upside_down_face:) could be spelled out: _nineteen hundred eighty-four_.

Or poke the [CMOS](https://www.chicagomanualofstyle.org/book/ed17/part2/ch09/toc.html) for more inspiration.

If I understand correctly, you are trying to implement the second option?

**Comment by dblock** [comment]
@mvz want to finish this? or should we merge as is?

**Comment by dblock** [comment]
@mvz can take a look on green

**Comment by mvz** [comment]
@dblock it's ready now as far as I'm concerned. I've renamed the option to `tens_of_hundreds`.

**Comment by jlduran** [comment]
This just needs an entry in the CHANGELOG, maybe also an addition to the README with the additional option. But that could be committed afterwards. I’ll leave it with @dblock. Thank you @mvz!

**Review by dblock — COMMENTED** [review]
@mvz care to please add to the change log/readme so we don’t forget? There’s no rush to merge. Thanks for contributing!

**Comment by jlduran** [comment]
If there are no objections, I can add the missing CHANGELOG entry this weekend and commit. Thank you!

**Pull request merged**